src/lib/replication/repl_dev/raft_state_machine.cpp

```diff
@@ -198,10 +198,21 @@ raft_buf_ptr_t RaftStateMachine::commit_ext(nuraft::state_machine::ext_op_params
     return m_success_ptr;
 }
 
+void RaftStateMachine::iterate_repl_reqs(std::function< void(int64_t, repl_req_ptr_t rreq) > const& cb) {
+    for (auto const& [key, rreq] : m_lsn_req_map) {
```

Comment:
I think its ok for iterator to be accessed by reference, given that m_lsn_req_map is a concurrenthashmap. Have you already checked out this https://github.com/facebook/folly/blob/main/folly/concurrency/ConcurrentHashMap.h#L29 

I just wanted to make sure its safe. If not we can do copy of it because they are cheap to do so.